### PR TITLE
Support true on-policy

### PR DESCRIPTION
**Pull request by fzyzcjy**
## Motivation

it is ready to use

plase merge https://github.com/sgl-project/sglang/pull/11958 before this one

logprob between train and rollout always zero:

image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/logits_processor.py [hunk 1]
```diff
@@ -593,6 +593,11 @@ def _get_logits(
                     None,  # bias
                     True,  # is_vnni
                 )
+            elif get_global_server_args().rl_on_policy_target == "fsdp":
+                # Due to tie-weight, we may not be able to change lm_head's weight dtype
+                logits = torch.matmul(
+                    hidden_states.bfloat16(), lm_head.weight.T.bfloat16()
```
Comment (hebiao064) [inline python/sglang/srt/layers/logits_processor.py:599]: why do we need to cast to bf16? i thought its already bf16

Comment (fzyzcjy) [reply python/sglang/srt/layers/logits_processor.py:599]: iirc (forget details) embedding must be fp32 to align w/ hf, then w/ weight tie the lm head is also fp32, but we want bf16 to align w/ hf here

Comment (hebiao064) [reply python/sglang/srt/layers/logits_processor.py:599]: is FSDP using BF16 for LM Head matmul? I thought FSDP will use FP32 for everything?

Comment (fzyzcjy) [reply python/sglang/srt/layers/logits_processor.py:599]: I forgot but matmul is bf16 at least
